compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -1105,8 +1114,11 @@ convertDataCon env m con args
                     [tmArg] -> pure tmArg
                     _ -> unhandled "constructor with more than two unnamed arguments" xargs
 
-            SimpleRecordCon ->
```

Comment:
IMO at this point we should split the handling of simple record types (these remain as `SimpleRecordCon`) and typeclass dictionaries (by adding a new `TypeClassDictCon` or something)